go-controller/pkg/cni/cniserver.go

```diff
@@ -102,6 +112,34 @@ func NewCNIServer(rundir string, useOVSExternalIDs bool, factory factory.NodeWat
 	return s, nil
 }
 
+// cancelPodAddsForOldMACs cancels and pod add requests that do not match
+// the given MAC address.
+func (s *Server) cancelPodAddsForOldMACs(pod *kapi.Pod, podMAC string) {
+	s.runningSandboxAddsLock.Lock()
+	defer s.runningSandboxAddsLock.Unlock()
+
+	for _, req := range s.runningSandboxAdds {
+		// Cancel sandbox requests for this pod that have a MAC set
+		// which doesn't match the expected MAC
+		if req.PodNamespace != pod.Namespace || req.PodName != pod.Name {
+			continue
+		}
+
+		updatedMAC, err := setMACUint64(podMAC, &req.UpdatedMAC)
```

Comment:
maybe I'm missing something but I dont see why you need all this MAC logic. Couldn't you just store the MAC in the pod request as a single field, then when you hit line 102, you just find and cancel the request with that old mac in it? I dont see why you need to store updatedMAC.

Reply:
It's stored in case the CNI ADD hasn't yet written the MAC to InitialMAC but the Update has been processed, since these things are async. Update may not trigger again to detect and kill the outdated ADD request.